app/models/user.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  MAXIMUM_NAME_LENGTH = 50
+  MAXIMUM_EMAIL_LENGTH = 250
+  VALID_REGEX_EMAIL = /\A[^.][\w\-?.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+[^.]\z/i
+  MINIMUM_PASSWORD_LENGTH = 6
+
+  before_save :email_to_downcase
```

Comment:
IMO it's better to format the email in a `before_validation` callback, that way it'll be ready for validation and saving.

Reply:
Thanks @violeta-p yes,  but I thought because the regex is "/i" case insensitive it would not be necessary ?

Reply:
Since you're already formatting the email in a `before_save` callback, IMO it's better to do so in `before_validation`, and with that the email will be formatted for validation and ready for saving. This is just my personal opinion. My colleagues are welcomed to share their opinion. :D

Reply:
`before_save` is fine but in case later on you want to add some custom validations you may encounter errors or simply end up changing it anyway. Up to you.